### ISSUE TITLE
Issues trying to train SqueezeNet

### ISSUE THREAD

**Issue by amandeepsaini** [leonardvandriel/caffe2_cpp_tutorial#23]
When I try to retrain the final layer of SqueezeNet, I get this error:

> terminate called after throwing an instance of 'caffe2::EnforceNotMet'
>   what():  [enforce fail at cross_entropy_op.cu:34] X.ndim() == 2. 4 vs 2 Error from operator: 
> input: "softmaxout" input: "label" output: "xent" type: "LabelCrossEntropy" device_option { device_type: 1 }

Similar error also arises when I try to train SqueezeNet from scratch. However, if I change the device option to cpu, the error changes slightly (not sure if it provides any insight, but here it is for reference):

> terminate called after throwing an instance of 'caffe2::EnforceNotMet'
>   what():  [enforce fail at accuracy_op.cc:10] X.ndim() == 2. 4 vs 2 Error from operator: 
> input: "softmax" input: "label" output: "accuracy" type: "Accuracy"
> 

GoogleNet, on the other hand, gives no issues while training or retraining, so it seems to be something specific to the architecture of SqueezeNet. Any ideas?

**Comment by leonardvandriel** [comment]
Hi @amandeepsaini,

Thanks for reporting this issue. The output of the SqueezeNet model has an incompatible shape for the accuracy operator. I updated the train and retrain example. You will need to run with `--reshape_output` in the case of SqueezeNet.

Let me know how that works.

**Comment by amandeepsaini** [comment]
Works great now, thanks!
